src/components/BasicChart.jsx

```diff
@@ -648,6 +625,111 @@ export default class BasicCharts extends React.Component {
+                                return 'inline';
+                            } else return null;
+                        })(),
+                    float: !config.legendOrientation ? 'right' : (() => {
+                        if (config.legendOrientation === 'left') return 'left';
+                        else if (config.legendOrientation === 'right') return 'right';
+                        else return null;
+                    })(),
+                }}
+            >
+                <VictoryLegend
+                    containerComponent={<VictoryContainer responsive />}
+                    centerTitle
+                    height={(() => {
+                        if (!config.legendOrientation) return this.state.height;
+                        else if (config.legendOrientation === 'left' || config.legendOrientation === 'right') {
+                            return this.state.height;
+                        } else return 100;
+                    })()}
+                    width={(() => {
+                        if (!config.legendOrientation) return 200;
+                        else if (config.legendOrientation === 'left' || config.legendOrientation === 'right') return 200;
+                        else return this.state.width;
+                    })()}
+                    orientation={
+                        !config.legendOrientation ?
+                            'vertical' :
+                            (() => {
+                                if (config.legendOrientation === 'left' || config.legendOrientation === 'right') {
+                                    return 'vertical';
+                                } else {
+                                    return 'horizontal';
+                                }
+                            })()
+                    }
+                    title="Legend"
+                    style={{
+                        title: { fontSize: 25, fill: config.legendTitleColor },
+                        labels: { fontSize: 20, fill: config.legendTextColor },
+                    }}
+                    data={legendItems.length > 0 ? legendItems : [{
+                        name: 'undefined',
+                        symbol: { fill: '#333' },
+                    }]}
+                    itemsPerRow={config.legendOrientation === 'top' || config.legendOrientation === 'bottom' ? 5 : null}
+                    events={[
+                        {
+                            target: 'data',
+                            eventHandlers: {
+                                onClick: config.interactiveLegend ? () => { // TODO: update doc with the attribute
+                                    return [
+                                        {
+                                            target: 'data',
+                                            mutation: (props) => {
+                                                console.info(props.index);
```

Comment:
I think we can remove the console info here also. If you want to log some information. we need to come up with proper logging framwork